utilities/generate_messages_catalog.py

```diff
@@ -131,96 +134,169 @@ def _is_callable(item):
         return False
 
 
-def _build_id_messages(python_package, package_root):
+def _find_modules_and_directories(top_level_directory):
+    """
+    Recursive helper function to find all python files included in top level
+    package. This will recurse down the directory paths of any package to find
+    all modules and subpackages in order to create an exhaustive list of all
+    python files within a given package.
+
+    :param top_level_directory: Path to the top level of a python package.
+    :type top_level_directory: str
+    :return: Returns a list of paths to all python files within that package.
+    :rtype: list [str]
+    """
+    modules = []
+    directories = []
+
+    for item in os.listdir(top_level_directory):
+        if item.endswith(".py"):
+            modules.append(os.path.join(top_level_directory, item))
+        elif os.path.isdir(os.path.join(top_level_directory, item)):
```

Comment:
like this here, but wondering, what if the dir you add here doesn't actually contain any python modules? would you still want it added to the output?